### DISCUSSION TITLE
Not able to see ActivityPub outbox on Mastodon

### DISCUSSION THREAD

**Discussion by markcellus** [mastodon/mastodon#17265] [Q&A] [answered]
I'm trying to get Mastodon to show the posts from my blog, `my-example.com` (pseudonym domain for privacy purposes). I'm exposing the blog entries using the ActivityPub protocol via an outbox, using the same instructions provided by the [How to implement a basic ActivityPub server](https://blog.joinmastodon.org/2018/06/how-to-implement-a-basic-activitypub-server/) guide for a user named `alice` (another pseudonym user for privacy purposes). 

However, when attempting to search for the `alice@my-example.com` user on a mastodon instance (https://mastodon.online), there are no results.

I've triple checked to make sure the webfinger URL, actor URL, etc are all returning the correct JSON and in the proper format Mastodon expects like the guide above instructs. The only thing that I didn't implement was the `inbox`. But since the `outbox` is implemented, I would still expect to see the blog entries show up in Mastodon search.

Is there something I'm missing here? Maybe there could be some implicit domain block on the mastodon instance that prevents it from seeing ActivityPub outboxes? Does `my-example.com` need to have a Mastodon instance installed on the server with an `alice` user registered (I hope not :confused: )? 

Thanks for any information anyone can provide.

**Comment by nightpool** [comment, marked as answer]
Hi Mark! Mastodon does not crawl any outboxes proactively, since this would consume a large amount of server resources for negligible benefit. Instead, we only show new posts created since *anyone on the local instance* has started following. If you follow the user `my-example.com` user from `mastodon.online`, and then make a new post, you should see it show up on Mastodon. this is because the primary way that users consume content on Mastodon is through their home feeds, so that's what our UX is built to accommodate. 

Note that you _will_ need to implement the `inbox` endpoint to allow users to follow you, and to be able to notify them of new posts (since you'll need to know *who* to send notifications for new posts to). This is probably what's stopping your profile from showing up at all—Mastodon requires the `inbox` endpoint to exist so that it's possible to interact with the user (follow them, like their posts, reply to their posts, etc)

Does that make sense?

**Reply by markcellus** [reply]
Ah yes that makes sense. So I would need to implement `inbox` and `following`  ActivityPub URLs for the `alice` user. Then follow the user on mastodon.social, then any new post from `alice`'s outbox from that point on will show up?

**Reply by nightpool** [reply]
Yes. The posts don't show up automatically when they appear in the `outbox`, though, you have to send the `Create` activity to the inbox of each following user. see https://www.w3.org/TR/activitypub/#delivery for more details

That is, ActivityPub is built on a pub-sub model of post delivery where servers need to proactively notify followers of their posts, rather then a polling based model where remote servers check for updates periodically
